.travis.yml

```diff
@@ -1,23 +1,34 @@
 language: python
 sudo: false
+services:
+  - docker
 python:
   - 2.7
   - 3.6
 env:
     - TEST_SERVER_MODE=false
     - TEST_SERVER_MODE=true
+before_install:
+  - export BOTO_CONFIG=/dev/null
 install:
-  - travis_retry pip install boto==2.45.0
-  - travis_retry pip install boto3
-  - travis_retry pip install .
-  - travis_retry pip install -r requirements-dev.txt
-  - travis_retry pip install coveralls==1.1
+  # We build moto first so the docker container doesn't try to compile it as well, also note we don't use
+  # -d for docker run so the logs show up in travis
   - |
+    python setup.py sdist
     if [ "$TEST_SERVER_MODE" = "true" ]; then
-      AWS_SECRET_ACCESS_KEY=server_secret AWS_ACCESS_KEY_ID=server_key moto_server -p 5000&
+      docker run --rm -t --name motoserver -e TEST_SERVER_MODE=true -e AWS_SECRET_ACCESS_KEY=server_secret -e AWS_ACCESS_KEY_ID=server_key -v `pwd`:/moto -p 5000:5000 -v /var/run/docker.sock:/var/run/docker.sock python:3.6.2-stretch /moto/travis_moto_server.sh &
```

Comment:
Is there a way we can use the current python version here instead of hardcoding to '3.6.2'? Currently we run in `SERVER_MODE=true` under all of the pythons listed in the `python:` YAML section above.

Maybe something like?:
```
python_version=$(python --version)
if [[ $python_version =~ " 3\." ]]; then
  docker_image="python:3.6.2-stretch"
else
  docker_image="python:????"
fi

docker run --rm -t --name motoserver \
  -e TEST_SERVER_MODE=true \
  -e AWS_SECRET_ACCESS_KEY=server_secret \
  -e AWS_ACCESS_KEY_ID=server_key \
  -v `pwd`:/moto -p 5000:5000 \
  -v /var/run/docker.sock:/var/run/docker.sock \
  ${docker_container} \
  /moto/travis_moto_server.sh &
```

Reply:
ya, actually looks like we can re-use an existing travis env var